artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -752,15 +752,12 @@ public RoutingStatus route(final Message message,
 
       Bindings bindings = addressManager.getBindingsForRoutingAddress(context.getAddress() == null ? message.getAddressSimpleString() : context.getAddress());
 
-      // TODO auto-create queues here?
-      // first check for the auto-queue creation thing
-      if (bindings == null) {
-         // There is no queue with this address, we will check if it needs to be created
-         //         if (queueCreator.create(address)) {
-         // TODO: this is not working!!!!
-         // reassign bindings if it was created
-         //            bindings = addressManager.getBindingsForRoutingAddress(address);
-         //         }
+      if (bindings == null && context.getAddress() != null) {
+         AddressSettings settings = server.getAddressSettingsRepository().getMatch(context.getAddress().toString());
+         if (settings.isAutoCreateQueues()) {
```

Comment:
@mtaylor Please review this.. this is fixing the compatibility between a 1.4 server and a 2.5 (snapshot) client.